charts/openfga/templates/job.yaml

```diff
@@ -1,4 +1,4 @@
-{{- if and (has .Values.datastore.engine (list "postgres" "mysql")) .Values.datastore.applyMigrations -}}
+{{- if and (has .Values.datastore.engine (list "postgres" "mysql")) .Values.datastore.applyMigrations .Values.datastore.waitForMigrations -}}
```

Comment:
We don't want the job only if `applyMigrations && waitForMigrations` . We want the job if `applyMigrations`, but we only want the initContainer if `applyMigrations && waitForMigrations`